cockpit/src/subscriptions-view.jsx

```diff
@@ -27,11 +27,61 @@ import { InsightsStatus } from './insights.jsx';
 let _ = cockpit.gettext;
 
 class Listing extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            attaching_in_progress: false,
+            attaching_in_progress_promise: null,
+            attaching_message: _("Auto-attach")
+        };
+        this.handleAutoAttach = this.handleAutoAttach.bind(this);
+        this.updateProgress = this.updateProgress.bind(this);
+    }
+    updateProgress(msg) {
+        this.setState({ attaching_message: msg });
+    }
+    handleAutoAttach(event) {
+        // only consider primary mouse button
+        if (!event || event.button !== 0)
+            return;
+        if (this.props.autoAttach) {
+            let self = this;
+            this.setState({
+                attaching_in_progress: true,
+                attaching_message: _("Auto-attaching ...")
+            });
+            this.props.autoAttach()
+                .done(function () {
+                    self.setState({
+                        attaching_in_progress: false,
+                        attaching_message: _("Auto-attach")
+                    });
+                })
+                .fail(function () {
+                    self.setState({
+                        attaching_in_progress: false,
+                        attaching_message: _("Auto-attach")
+                    });
+                })
+                .progress((message) => {
```

Comment:
Can you please send me any link to documentation/article, why it should not be used and what should be used instead?

Reply:
(I don't think we have docs, and I realize that I might be ahead of the team here with my ideas of deprecation... I made a note to write something up early next decade.)

The `progress` handlers are the counter part to `notify`; if you call `notify` on a Deferred object, all its promises get their `progress` handlers called.  This is neat, but standard JavaScript promises don't support this.  So if you mix cockpit promises with standard promises (which ordinarily would work) progress reporting will break.

Also, in this case, I think nobody is calling `notify`, so the `progress` handler will never be called, either.  It can just be removed.

Reply:
And if you want progress reporting, I would propose to add a `update_progress` parameter to `client.autoAttach`.